backend/src/v2/objectstore/object_store.go

```diff
@@ -324,15 +319,24 @@ func getMinioCredential(ctx context.Context, clientSet kubernetes.Interface, nam
 		minioArtifactSecretName,
 		metav1.GetOptions{})
 	if err != nil {
-		return cred, err
+		return nil, err
 	}
-	cred.AccessKey = string(secret.Data["accesskey"])
-	cred.SecretKey = string(secret.Data["secretkey"])
-	if cred.AccessKey == "" {
-		return cred, fmt.Errorf("does not have 'accesskey' key")
+	accessKey := string(secret.Data["accesskey"])
+	secretKey := string(secret.Data["secretkey"])
+
+	if accessKey != "" && secretKey != "" {
+		cred = credentials.NewStaticCredentials(accessKey, secretKey, "")
+		return cred, err
 	}
-	if cred.SecretKey == "" {
-		return cred, fmt.Errorf("does not have 'secretkey' key")
+
+	aws_cred, err := getAWSCredential()
+	if aws_cred != nil {
+		return aws_cred, err
 	}
-	return cred, nil
+
+	return nil, fmt.Errorf("does not have 'accesskey' or 'secretkey' key")
+}
+
+func getAWSCredential() (cred *credentials.Credentials, err error) {
+	return credentials.NewCredentials(&credentials.ChainProvider{}), nil
```

Comment:
@ryansteakley are you sure that you did not forget to specify the providers in the `ChainProvider`?

Because by default a chain provider has no internal providers and will always error.